card/src/main/java/com/adyen/checkout/card/data/CardBrand.kt

```diff
@@ -0,0 +1,52 @@
+package com.adyen.checkout.card.data
+
+import android.os.Parcel
+import android.os.Parcelable
+
+data class CardBrand(val txVariant: String) : Parcelable {
+
+    internal var cardType: CardType? = null
+        private set
+
+    /**
+     * Use this constructor when defining the supported card type predefined inside [CardType] enum
+     * inside your component
+     */
+    constructor(cardType: CardType) : this(cardType.txVariant) {
+        this.cardType = cardType
+    }
+
+    private constructor(parcel: Parcel) : this(parcel.readString().orEmpty())
+
+    override fun describeContents(): Int = 0
+
+    override fun writeToParcel(parcel: Parcel, i: Int) {
+        parcel.writeString(txVariant)
+    }
+
+    companion object {
+        @JvmField
+        val CREATOR = object : Parcelable.Creator<CardBrand> {
+            override fun createFromParcel(parcel: Parcel): CardBrand {
+                return CardBrand(parcel)
+            }
+
+            override fun newArray(size: Int): Array<CardBrand?> {
+                return arrayOfNulls(size)
+            }
+        }
+
+        /**
+         * Get [CardBrand] from the brand name as it appears in the Checkout API.
+         */
+        @JvmStatic
+        fun getByBrandName(brand: String): CardBrand {
```

Comment:
Why do we still have this method?

Reply:
Because we still sometimes need to compare some card types with the ones inside `CardType` so we need to map brands coming from checkout api to the corresponding `CardType` if needed.

Reply:
For example in `CardView` line 278

Reply:
Yes but `CardBrand` has a `cardType` filed inside it and can be initialized with a string `txVariant`, shouldn't this be enough to achieve what we want?

Reply:
To explain more, with the current code `CardBrand("visa").cardType == null` but  `CardBrand. getByBrandName("visa").cardType == CardType.VISA`. Can you see the issue with this method better now?